### PR TITLE
fix path resolution for writeFile in nodefs

### LINKED ISSUE #10150: fs.writeFileSync and relative path incorrect on windows
### What version of Bun is running?

1.1.3+2615dc742

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

Using latest version of bun for windows (v1.1.3)
**fs.writeFileSync** and relative paths do not work as expected (does not work like nodejs)

`fs.mkdirSync('../../config')`
=> OK

`fs.writeFileSync('../../test.txt', data)`
=> file will be created in **wrong directory**, will always be `../test.txt` regardless of `../../../..`

### What is the expected behavior?

_No response_

### What do you see instead?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by gvilums**
### What does this PR do?
Fixes #10150

### How did you verify your code works?
Add a new test that checks this case

### CODE REVIEW

File: src/bun.js/node/node_fs.zig [hunk 1]
```diff
@@ -5336,58 +5336,12 @@ pub const NodeFS = struct {
...
-                if (Environment.isWindows) {
-                    // windows openat does not support path traversal, fix it here.
-                    // use pathbuf2 here since without it 'panic: @memcpy arguments alias' triggers
-                    if (std.mem.indexOf(u8, path, "\\.\\") != null or std.mem.indexOf(u8, path, "\\..\\") != null) {
-                        const fixed_path = bun.path.normalizeStringWindows(path, &pathbuf2, false, false);
-                        pathbuf2[fixed_path.len] = 0;
-                        path = pathbuf2[0..fixed_path.len :0];
-                    }
-                }
+                const path = args.file.path.sliceZ(pathbuf);
```
Comment (dylan-conway) [inline src/bun.js/node/node_fs.zig]: Before this was using `sliceZWithForceCopy`. If the path isn't copied into the buffer and an error is returned after `openat`, will the path memory in the error still be alive everywhere it's used?

Comment (Jarred-Sumner) [reply src/bun.js/node/node_fs.zig]: yeah i think this makes tests flakier

Comment (gvilums) [reply src/bun.js/node/node_fs.zig]: updated to use force copy

**Comment by Jarred-Sumner** [comment]
@gvilums test is failing
```js
2986 | it("using writeFile on an fd does not truncate it", () => {
2987 | const temp = tmpdir();
2988 | const fd = fs.openSync(join(temp, "file.txt"), "w+");
^
EPERM: Operation not permitted
errno: -1
syscall: "open"
path: "C:\Users\RUNNER~1\AppData\Local\Temp\bun-test-tmp-7l0d_-734694376_61zg\file.txt"
```

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**